pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +270,27 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns a list of users encoded in SPIFFE format with the local
+// trust domain and its aliases.
+// For example, if the local trust domain is "cluster.local", and its aliases are "td1", "td2".
+// For a user "bar" in namespace "foo", getPrincipalsIncludingAliases returns
+// cluster.local/ns/foo/sa/bar, td1/ns/foo/sa/bar, td2/ns/foo/sa/bar
+// The trust domain corresponds to the trust root of a system.
+// Refer to
+// [SPIFFE-ID](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain)
+// In Istio, an identity is presented in the SPIFFE format, which is:
+// spiffe://<trust-domain>/ns/<some-namespace>/sa/<some-service-account>
+func getPrincipalsIncludingAliases(trustDomainAliases []string, principals []string) []string {
+	usersWithAliases := principals
+	for _, tdAlias := range trustDomainAliases {
+		for _, principal := range principals {
+			newTrustDomainIdentity, err := getIdentityWithNewTrustDomain(tdAlias, principal)
```

Comment:
Do you need to check if the original trust domain is part of the `trustDomainAliases` before this expansion?

Reply:
@diemtvu I don't think there is such need (to complicate this function)? Users should not put the original trust domain to the alias list. Even if they do, we can still generate one extra string and it should not affect perform much(?) and RBAC policy won't be affected since it's addictive. Wdyt?

@liminw I'm not sure I understand your comment. Why is "cluster.local" special here? I thought we agreed we should treat it as just another string with no special value.

Reply:
@pitlv2109 Please see [the relevant part of the design doc ](https://docs.google.com/document/d/1ncrENtF6PBVZVjhYePbvZ6F1obbNApKJh5mUDRvwHMw/edit#bookmark=id.r2z0eop9dwmr).

"cluster.local" is a reference to local trust domain, which is set by "trust_domain" and "trust_domain_aliases" fields. This is my comment #1 above (i..e, expand "cluster.local" based on trustDomainAliases, but not include "cluster.local" string). 

Ideally we should only see "cluster.local" in authorization policies if all accesses are within the same trust domain. However, for backwards compatibility, we still need to take into account the hard coded trust domain cases like "td1/ns/foo/sa/bar", in which case, if trustDomainAliases contains ["td1", "td2"], the identity string can be expanded to two strings "td1/ns/foo/sa/bar", and "td2/ns/foo/sa/bar". This is my comment #2 above. You must check if the trust domain part of the identity string ("td1" in the example) is in the trustDomainAliases ("td1" and "td2" in the example) before you apply the expansion.

Reply:
It could also be a bug if you not do. Let's say, for some reasons, I have the authz that use TD `foo`, even though `foo` is not the TD nor TD aliases. Without checking, such dummy authZ rule may grant access by accident.

Reply:
@liminw But I thought we agreed to treat "cluster.local" as another string (i.e. not thing special about it) when we discussed it offline with @diemtvu. Why do we have to treat it with special effect anyway? I think it can complicate the code, as well as the UX.

Reply:
@diemtvu I see what you mean now, but then again, we will need to check if the existing trust domain in the policy is in the original trust domain or not, too, which is a problem outside of this PR (regardless of existence of the aliases). I can file an issue and a follow up PR for that matter.

Reply:
To be exact, it's the problem because of this PR :). If alias is not used, you use authZ policy as is, which is fine. But when aliases are available, you don't want to treat foo the same as bar, if foo and bar are not aliases. You should expand the authz rule only for those in the aliase group (doesn't need to be the main TD).

To be clear, let's say the TD is `bar`, and aliase is `baz`

Then a authZ rule on `bar` (or `baz`) will be cloned into 2 rules, one for `bar`, one for `baz`
But a rule on `foo` should stay the same.

Reply:
Actually, the previous code has some bug. I revised the code and checked if the extracted trust domain from the existing policy is part of the aliases. If so, it will generate config for the current trust domain and the aliases.